### DISCUSSION TITLE
Pass extra variables to overriden template

### DISCUSSION THREAD

**Discussion by TheoStefou** [ckan/ckan#7451] [Q&A]
Given that I have overriden the template package/search.html, how can I pass extra variables to it? I would like my template to have some extra variables that I have calculated myself in my plugin.

The only way I have found to do this so far is by creating a helper function that returns the value, but this seems hacky.

toolkit.render doesn't seem to be an option, as it doesn't do anything to make sure the overriding template receives the required variables of the parent template (e.g. using toolkit.render with extra_vars= a dict with my extra variables, yields errors of the sort: "dataset_type" key error and so on)

**Comment by pdelboca** [comment]
Hello @TheoStefou using helpers is a viable options. Helpers in CKAN are the equivalent of Flask's [context_processors](https://flask.palletsprojects.com/en/2.2.x/templating/#context-processors) (in functionality) so they are safe to use.

Since templates are rendered in core Blueprints we do not have another another official way to send extra variables to it. 

Adding variables to the `g` object has been used in the past. We are no longer using it in core but it could be another option for your usecase.
